src/actions/wapeSale/purchaseDomains.ts

```diff
@@ -0,0 +1,121 @@
+import * as ethers from "ethers";
+import { getSaleStatus } from ".";
+import { WapeSale } from "../../contracts/types";
+import { Claim, SaleStatus, Mintlist, Maybe } from "../../types";
+
+const abi = ["function masterCopy() external view returns (address)"];
+
+const errorCheck = async (condition: boolean, errorMessage: string) => {
```

Comment:
Do we need `async` in this function? @MattCharles

Reply:
no, but if there's no await, i don't think the transpiler will spit out a promise anyway.

Reply:
I would remove the unnecessary async here regardless.

Reply:
at this point we're definitely at the "merge anything that's not utterly egregious" stage of the process. haha.

Reply:
Fair lol